src/components/alert/index.js

```diff
@@ -0,0 +1,43 @@
+import { Alert } from "bootstrap";
+import BaseComponent from "../base-component";
+
+export default class CustomAlert extends BaseComponent {
+  timerId;
+  constructor(type = "success", message = "") {
+    super();
+    this.type = type;
+    this.message = message;
+
+    this.init();
+
+    // NOTE: register alert for closing
+    new Alert(this.element);
+
+    this.timerId = setTimeout(() => {
+      this.close();
+    }, 2000);
```

Comment:
**Performance Issue**: The use of 'setTimeout' for managing alert visibility can lead to performance issues and memory leaks if not properly cleared when the component is destroyed or when multiple alerts are triggered in rapid succession. <br> **Fix**: Implement a more robust mechanism for handling alert visibility, possibly using requestAnimationFrame for better performance and control. Ensure that all timeouts are cleared when the alert component is destroyed or re-initialized. <br> **Code Suggestion**: 
 ```
 - this.timerId = setTimeout(() => {
 -   this.close();
 - }, 2000);
 + this.timerId = requestAnimationFrame(this.close.bind(this));
 ```